components/core/src/clp/streaming_compression/lzma/Compressor.cpp

```diff
@@ -0,0 +1,205 @@
+#include "Compressor.hpp"
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+
+#include <lzma.h>
+#include <spdlog/spdlog.h>
+
+#include "../../ErrorCode.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../../type_utils.hpp"
+
+namespace clp::streaming_compression::lzma {
+auto Compressor::open(FileWriter& file_writer) -> void {
+    if (nullptr != m_compressed_stream_file_writer) {
+        throw OperationFailed(ErrorCode_NotReady, __FILENAME__, __LINE__);
+    }
+
+    m_lzma_stream.detach_input();
+    if (false
+        == m_lzma_stream.attach_output(
+                m_compressed_stream_block_buffer.data(),
+                m_compressed_stream_block_buffer.size()
+        ))
+    {
+        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
```

Comment:
This is because the output buffer is initialized internally by the LZMA compressor instead of being passed in by the user.
If the output buffer args are invalid, it is due to incomplete initialization

Reply:
hmm, my understanding is that NotInit should work for other methods such as Write() or flush(), so I am persoanlly leaning towards  ErrorCode_NoMem or ErrorCode_Failure.
Let's get some input from senior people @kirkrodrigues @davidlion

Reply:
Let's use `ErrorCode_NoMem`

Reply:
@kirkrodrigues Maybe you can give a final say on this error code choice.

Reply:
I would use `ErrorCode_Failure` since this condition shouldn't only be hit if there's a programming error. `ErrorCode_NoMem` seems misleading since if we truly couldn't allocate the buffer, then the code shouldn't even get here.

Reply:
As discussed, please use ErrorCode_Failure.